decidim-proposals/app/commands/decidim/proposals/admin/publish_participatory_text.rb

```diff
@@ -39,14 +39,37 @@ def call
 
         def publish_drafts
           Decidim::Proposals::Proposal.where(component: form.current_component).drafts.find_each do |proposal|
-            add_failure(proposal) unless proposal.update(published_at: Time.current)
+            add_failure(proposal) unless publish_proposal(proposal)
           end
           raise ActiveRecord::Rollback if @failures.any?
         end
 
         def add_failure(proposal)
           @failures[proposal.id] = proposal.errors.full_messages
         end
+
+        # This will be the PaperTrail version shown in the version control feature (1 of 1).
+        # For an attribute to appear in the new version it has to be reset
+        # and reassigned, as PaperTrail only keeps track of object CHANGES.
+        def publish_proposal(proposal)
+          title, body = reset_proposal_title_and_body(proposal)
+
+          Decidim.traceability.perform_action!(:create, proposal, form.current_user, visibility: "all") do
```

Comment:
Maybe the action here should be `publish` instead of `create` and we could avoid these "hacks"?

Reply:
I'm not sure I follow.
This PR does not affect the way the action_log is being rendered. It keeps the old behaviour:

![participatory_text_admin_log](https://user-images.githubusercontent.com/40306853/66314821-e27d7000-e914-11e9-8622-46b74290e68b.png)

But I agree that the code smells.
The issue is having only one version after a proposal is published and to reflect the state of the model just before publishing... I don't see a better solution right now, but I'm open to suggestions.

Reply:
Sorry, you're right. I just thought that since you're publishing a proposal, not creating it, the action could be different. But it has nothing to doo with papertrail issues.